nixos/modules/installer/tools/nixos-option/nixos-option.cc

```diff
@@ -588,13 +588,14 @@ int main(int argc, char ** argv)
     std::string optionsExpr = "(import <nixpkgs/nixos> {}).options";
     std::string configExpr = "(import <nixpkgs/nixos> {}).config";
     std::vector<std::string> args;
+    std::string_view pname = nix::baseNameOf(argv[0]);
 
     struct MyArgs : nix::LegacyArgs, nix::MixEvalArgs
     {
         using nix::LegacyArgs::LegacyArgs;
     };
 
-    MyArgs myArgs(nix::baseNameOf(argv[0]), [&](Strings::iterator & arg, const Strings::iterator & end) {
+    MyArgs myArgs(pname.data(), [&](Strings::iterator & arg, const Strings::iterator & end) {
```

Comment:
Perhaps making `nix::LegacyArgs::LegacyArgs` accept `std::string_view` is a cleaner idea.